Docs/1164 toggle group documentation (#1602)
sl-design-system/components: website/src/categories/components/toggle-group/usage.md

</sl-toggle-group>
</div>

<div class="ds-code">

  ```html
    <sl-toggle-group multiple>
      <sl-toggle-button aria-label="Bold" pressed>
        <sl-icon name="far-bold" slot="default"></sl-icon>
        <sl-icon name="fas-bold" slot="pressed"></sl-icon>
      </sl-toggle-button>
      <sl-toggle-button aria-label="Italic">
        <sl-icon name="far-italic" slot="default"></sl-icon>
        <sl-icon name="fas-italic" slot="pressed"></sl-icon>
      </sl-toggle-button>
      <sl-toggle-button aria-label="Underline" pressed>
        <sl-icon name="far-underline" slot="default"></sl-icon>
        <sl-icon name="fas-underline" slot="pressed"></sl-icon>
      </sl-toggle-button>
    </sl-toggle-group>
    <sl-toggle-group>
      <sl-toggle-button>All types</sl-toggle-button>
      <sl-toggle-button>Primary education</sl-toggle-button>
      <sl-toggle-button pressed>Secondary education</sl-toggle-button>
    </sl-toggle-group>
  ```

</div>

</section>

<section>

## When to use
Toggle group are best used in situations where users need to make a single selection from multiple options within a defined group. Here are two common scenarios:

### Related Exclusive Choices

> Use a toggle group when offering a set of options where only one can be active at a time. This is ideal for scenarios like mode selection, where choosing one option automatically deselects others (e.g., light vs. dark mode). The toggle group ensures clear feedback, allowing users to easily understand which option is currently active.

Reviewer: We should make clearer that this is just a visual change; if you have the same data but it looks differently you can use a toggle group, if you load different data altogether you need to use tabs or navigation links.

### Multiple of Related Actions
A toggle group is useful when presenting a series of actions that are related but independent, where users can activate one, some, or all options at once. This pattern works well for scenarios like turning on different filters or settings, where multiple combinations of active toggles are allowed. The visual grouping helps users understand the relationship between these actions.

> ### Requirer Immediate Feedback

Reviewer: ```suggestion
### Require Immediate Feedback
```

### Unrelated Actions
Avoid using a toggle group when the actions or options are unrelated and do not influence each other. If each option serves a completely different purpose or affects different parts of the interface, placing them in a toggle group may confuse users into thinking they are interconnected. In such cases, it's better to present them as separate controls to make their independence clearer.

### Persistent Interactions
If the user selections need to persist across different interactions or sessions. Toggle groups are best suited for actions that require immediate feedback and may not be suitable for settings that need to be saved long-term. For more persistent states, consider using checkboxes, radio buttons, or a settings menu that ensures the selections remain even after the user navigates away or reloads the page.

</section>

<section>

## Anatomy

|Item|Name| Description | Optional|
|-|-|-|-|
|1|Toogle Button |Description |Yes|
|2|Container |Description |Yes|
|3|Icons |Description |No|
|3|Label |Description |No|

{.ds-table}

</section>

<section>

## Sizing

> Toggle Group come in two sizes, to match diferents scenarios:

Reviewer: ```suggestion
Toggle Group come in two sizes, to match diferent scenarios:
```

- **Medium:** The medium-sized toggle group function as the default option across our user interfaces. It achieves a balanced blend of size and clarity, establishing itself as the standard and user-friendly choice for single-choice.

- **Large:** Choosing a large toggle group for touch-based devices offers several advantages, as it provides a larger tap target that improves accessibility and ease of interaction. This larger size is especially beneficial in important or high-priority scenarios, where ensuring precise user input is critical.

</section>

<section>

## Status

> Toggle Group come in two sizes, to match diferents scenarios:

Reviewer: ```suggestion
Buttons in a toggle group have 3 different states, to match diferent scenarios:
```

- **Pressed:** The toggle is in an selected state, signifying that the corresponding action is currently applied or in effect. This state provides clear feedback that the user’s input has been recognized.

- **Disabled:** The toggle is inactive and cann't be interacted with, typically because the options are unavailable or restricted in the current context. This state visually indicates that no action can be performed on the toggle at this time.

</section>

<section>

## Variants

Toggle Group come in various versions, each suited for specific situations:

- **Icon (Default):** This variant displays only an icon, providing a compact visual representation of the action, ideal for limited space or familiar actions.

- **Label:** This variant shows just a text label, offering clear and direct descriptions of the action, best for situations where clarity is prioritized.

- **Icon & Label:** TThis variant combines both an icon and a label, delivering both visual and textual cues to enhance clarity and recognition, ideal for important or less familiar actions.

</section>

<section>

## Figma options

> With these options, you can tweak the appearance of the radio in Figma. They are available in the Design Panel so you can compose the toggle group to exactly fit the user experience need for the use case you are working on.

Reviewer: ```suggestion
With these options, you can tweak the appearance of the toggle group in Figma. They are available in the Design Panel so you can compose the toggle group to exactly fit the user experience need for the use case you are working on.
```

|Item|Options|Description|
|-|-|-|
|Size|`'md', 'lg'`|The button is available in three sizes. If not specified the default value is `md` .|

> |Segments|`'5'` to `'2'`|The toggle group offers the possibility to display fron 5 to 2 controls|

Reviewer: ```suggestion
|Segments|`'5'` to `'2'`|The toggle group offers the possibility to display from 5 to 2 controls|
```
5 to 2 is a bit weird. Why not `2 to 5`?

|Item|Name| Description | Optional|
|-|-|-|-|

> |1|Toogle Button |These are the control button, for more information check the **toggle group**. |Yes|

Reviewer: ```suggestion
|1|Toggle Button |These are the control buttons, for more information check the **toggle group**. |Yes|
```
Should this not say "toggle button" at the end? And then link to the toggle button page?
